test/e2e/atlas/search_nodes_test.go

```diff
@@ -0,0 +1,200 @@
+	g.tier = tierM10
+	g.mDBVer = minSearchNodesMDBVersion
+	g.generateCluster()
+
+	t.Run("Verify no search node setup yet", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			clustersEntity,
+			searchEntity,
+			nodesEntity,
+			"list",
+			"--clusterName", g.clusterName,
+			"--projectId", g.projectID,
+			"-o=json",
+		)
+
+		require.Error(t, cmd.Run())
+	})
+
+	t.Run("Create search node", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			clustersEntity,
+			searchEntity,
+			nodesEntity,
+			"create",
+			"--clusterName", g.clusterName,
+			"--projectId", g.projectID,
+			"--file", "search_nodes_spec.json",
+			"-w",
+			"-o=json",
+		)
+
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		resp = bytes.TrimLeft(resp, ".")
+
+		require.NoError(t, err, resp)
+		var searchNode atlasv2.ApiSearchDeploymentResponse
+		require.NoError(t, json.Unmarshal(resp, &searchNode))
+
+		assert.Equal(t, *searchNode.GroupId, g.projectID)
+		assert.Equal(t, searchNode.Specs, pointer.Get([]atlasv2.ApiSearchDeploymentSpec{
+			{
+				InstanceSize: "S30_LOWCPU_NVME",
+				NodeCount:    2,
+			},
+		}))
+	})
+
+	t.Run("List + verify created", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			clustersEntity,
+			searchEntity,
+			nodesEntity,
+			"list",
+			"--clusterName", g.clusterName,
+			"--projectId", g.projectID,
+			"-o=json",
+		)
+
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		resp = bytes.TrimLeft(resp, ".")
+
+		require.NoError(t, err, resp)
+		var searchNode atlasv2.ApiSearchDeploymentResponse
+		require.NoError(t, json.Unmarshal(resp, &searchNode))
+
+		assert.Equal(t, *searchNode.GroupId, g.projectID)
+		assert.Equal(t, searchNode.Specs, pointer.Get([]atlasv2.ApiSearchDeploymentSpec{
+			{
+				InstanceSize: "S30_LOWCPU_NVME",
+				NodeCount:    2,
+			},
+		}))
+	})
+
+	t.Run("Update search node", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			clustersEntity,
+			searchEntity,
+			nodesEntity,
+			"update",
+			"--clusterName", g.clusterName,
+			"--projectId", g.projectID,
+			"--file", "search_nodes_spec_update.json",
+			"-w",
+			"-o=json",
+		)
+
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		resp = bytes.TrimLeft(resp, ".")
+
+		require.NoError(t, err, resp)
+		var searchNode atlasv2.ApiSearchDeploymentResponse
+		require.NoError(t, json.Unmarshal(resp, &searchNode))
+
+		assert.Equal(t, *searchNode.GroupId, g.projectID)
+		assert.Equal(t, searchNode.Specs, pointer.Get([]atlasv2.ApiSearchDeploymentSpec{
+			{
+				InstanceSize: "S20_HIGHCPU_NVME",
+				NodeCount:    3,
+			},
+		}))
+	})
+
+	t.Run("List + verify updated", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			clustersEntity,
+			searchEntity,
+			nodesEntity,
+			"list",
+			"--clusterName", g.clusterName,
+			"--projectId", g.projectID,
+			"-o=json",
+		)
+
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		resp = bytes.TrimLeft(resp, ".")
+
+		require.NoError(t, err, resp)
```

Comment:
```suggestion
		require.NoError(t, err, resp)
```

do we need the TrimLeft here?

Reply:
Yes, see other comments